xinference/model/llm/pytorch/intern_vl.py

```diff
@@ -346,7 +363,121 @@ def chat(
                 )
             ],
             usage=CompletionUsage(
-                prompt_tokens=-1, completion_tokens=-1, total_tokens=-1
+                prompt_tokens=prompt_tokens,
+                completion_tokens=completion_tokens,
+                total_tokens=prompt_tokens + completion_tokens,
+            ),
+        )
+        return chunk
+
+    def _generate_stream(
+        self,
+        pixel_values,
+        content,
+        generation_config,
+        num_patches_list,
+        history,
+        include_usage,
+    ):
+        from threading import Thread
+
+        from transformers import TextIteratorStreamer
+
+        # Initialize the streamer
+        streamer = TextIteratorStreamer(
+            self._tokenizer, skip_prompt=True, skip_special_tokens=True, timeout=10
+        )
+        # Define the generation configuration
+        generation_config["streamer"] = streamer
+        # Start the model chat in a separate thread
+        thread = Thread(
+            target=self._model.chat,
+            kwargs=dict(
+                tokenizer=self._tokenizer,
+                pixel_values=pixel_values,
+                question=content,
+                num_patches_list=num_patches_list,
+                history=history,
+                return_history=False,
+                generation_config=generation_config,
             ),
         )
-        return self._to_chat_completion(chunk)
+        thread.start()
+
+        completion_id = str(uuid.uuid1())
+        prompt_tokens, completion_tokens, total_tokens = 0, 0, 0
+        prompt_tokens = self._get_input_tokens(content, history, num_patches_list)
```

Comment:
Is it possible that we generate the prompt, and turn it into token ids, so we can calculate the prompt_tokens, the benefit is that we don't need to tokenize twice.